lib/ansible/module_utils/network/common/cfg/base.py

```diff
@@ -13,6 +13,12 @@
 class ConfigBase(object):
     """ The base class for all resource modules
     """
+    ACTION_STATES = ['merged', 'replaced', 'overridden', 'deleted']
+
     def __init__(self, module):
         self._module = module
-        self._connection = get_resource_connection(module)
+        self.state = module.params['state']
+        self._connection = None
+
+        if self.state not in ['rendered', 'parsed']:
```

Comment:
I'm curious why we need to change the base cfg to support something like this? This would be the first module to do so.

Reply:
These changes are to support new states changes .